### ISSUE TITLE
Blacklist - Hide contributions of user

### ISSUE THREAD

**Issue by roschaefer** [Human-Connection/WebApp#211]
**Is your feature request related to a problem? Please describe.**
Not every post is everybody's cup of tea. Sometimes a user dislikes posts by a certain person so much, that he/she would like to stop seeing contributions of this person.

**Describe the solution you'd like**
In the case above, a user should be able to click on the contribution or the authors profile page and click a button to *blacklist* this person and stop seeing contributions of this person.

**Describe alternatives you've considered**
A possible alternative could be to block accounts for the entire community.

**Additional context**
```gherkin
Feature: Individual Blacklist
  As a user
  I want to click on a button to blacklist certain user profiles
  In order to stop seeing contributions of this account, because I don't like them
```

**Comment by roschaefer** [comment]
Should the button be placed on the profile page, the contributions page or both?

**Comment by sebako86** [comment]
@roschaefer both would be good

**Comment by appinteractive** [comment]
Should be on both. 

Other question: Should we completely hide the contributions or maybe just mask them to show that there is a contribution/comment from an blocked user so that we could unblock/show that contribution on time? At least for comments that could make sense or not?

**Comment by roschaefer** [comment]
I vote for a strict filter - if I blacklisted somebody I mean it.
However I would like to unblock people. If I can't see their contributions, I cannot undo my decision the same way. Therefore a unblock feature would make sense on my profile page.

**Comment by sebako86** [comment]
+1 for a strict filter. 

To unblock we'd need a list like Organizations in Settings/Profile with all blocked peers listed.

**Comment by appinteractive** [comment]
But on comments some hint would be nice to understand reactions from others don't you think?

**Comment by sebako86** [comment]
true that, could be confusing if there's an answer but no question. Maybe we should label it as a riddle 🍡

**Comment by roschaefer** [comment]
What is the umbrella term for contributions and comments?

**Comment by appinteractive** [comment]
>What is the umbrella term for contributions and comments? ...

@roschaefer userContent?

**Comment by roschaefer** [comment]
What is the expected response if the user visits a contribution site that is blacklisted? With the [current implementation](https://github.com/Human-Connection/API/pull/124) I get `404`. Another user who does not blacklist the author will see the contribution. Is that expected behaviour?

**Comment by sebako86** [comment]
If someone uses a bookmark to open some post from a meanwhile blocked user I'd say redirect to newsfeed, any issues with that?

**Comment by roschaefer** [comment]
I would prefer 404 or an explicit message telling you that you blacklisted the author. I don't like redirect because of magic.

**Comment by sebako86** [comment]
true that. Hmm how about showing the contribution concealed (blur filter or smth.) with a note on top?

**Comment by sushidave** [comment]
How about calling it _unsubscribe_ instead of _blocking_? While technically it does the same, _unsubscribe_ sounds less "violent".

Examples: 
- "Unsubscribe from this user"
- "Unsubscribe from comments by this user"
- "Unsubscribe from contributions by this user"

**Comment by sushidave** [comment]
Refering to my previous post I probably mixed up things. Twitter has three kinds of "disconnecting" features:

1. [Unfollow](https://help.twitter.com/en/using-twitter/how-to-unfollow-on-twitter)
2. [Mute](https://help.twitter.com/en/using-twitter/twitter-mute)
3. [Block](https://help.twitter.com/en/using-twitter/blocking-and-unblocking-accounts)

Each one has its own consequences. And there are also features to report a users for violations.

@roschaefer Is your feature request rather about no. 2 (muting) or about no. 3 (blocking)?

**Comment by appinteractive** [comment]
@sushidave as I know it’s more no 3

**Comment by mimicc83** [comment]
@sushidave it's intended for blocking (total removal from your sight)

**Comment by roschaefer** [comment]
@appinteractive @mimicc83 really? I thought it's more like muting.

**Comment by appinteractive** [comment]
To be honest I don’t like that feature to be blocking. But currently we have no subscriptions beside „following“ a user or organization for updates. The rest is „take it all“. So maybe we could name it „unsubscribe“ to be more friendly. What do you thing? What would a „unsubscribe“ look like?

**Comment by roschaefer** [comment]
I thought "unsubscribe === unfollow" and this feature here is to "mute" sb.

There is no blocking because we don't have direct messages implemented.

**Comment by appinteractive** [comment]
But it hides the content of the blacklisted users!? I don’t get the subtle differences between the three types. 

But the purpose of that issue was to give people to hide content from users that annoyed them.

**Comment by mimicc83** [comment]
yes you're right, unfollow === unsubscribe - mute or whatever name we use should also work for chat I think. I wouldn't necessarily call it "block" and rather stick to mute as it sounds more nicey. Can we stick to one for now (mute) and discuss naming at some later point? I think the functionality required is clear to everyone.
